[Windows] graphapi gtests are not compiling due to missing class methods

(create on behalf of @apwojcik ).

After #2917 , the graphapi gtests are not compiling on Windows. It seems unrelated to the Windows and compiler stack because the error reported concerns missing class methods.

```
[ 76%] Built target test_deepbench_gru
In file included from C:\UIF\MIOpen\test\gtest\execute_primitive.cpp:28:
In file included from C:/UIF/MIOpen/src/include\miopen/execution_context.hpp:32:
C:/UIF/MIOpen/src/include\miopen/sqlite_db.hpp:519:51: error: no member named 'WhereClause' in 'TestProblemDescription'
  519 |         std::tie(clause, values) = problem_config.WhereClause();
      |                                    ~~~~~~~~~~~~~~ ^
C:/UIF/MIOpen/src/include\miopen/sqlite_db.hpp:352:50: note: in instantiation of function template specialization 'miopen::SQLitePerfDb::RemoveUnsafe<TestProblemDescription>' requested here
  352 |         return reinterpret_cast<Derived*>(this)->RemoveUnsafe(args...);
      |                                                  ^
C:/UIF/MIOpen/src/include\miopen/find_solution.hpp:77:17: note: in instantiation of function template specialization 'miopen::DbTimer<miopen::MultiFileDb<miopen::SQLitePerfDb, miopen::SQLitePerfDb, true>>::Remove<TestProblemDescription, std::basic_string<char>>' requested here
   77 |         if(db().Remove(problem, s.SolverDbId()))
      |                 ^
C:/UIF/MIOpen/src/include\miopen/find_solution.hpp:191:9: note: in instantiation of function template specialization 'miopen::solver::FindSolutionImpl<TunableTestSolver, miopen::ExecutionContext, TestProblemDescription, (lambda at C:\UIF\MIOpen\src\include\miopen\find_solution.hpp:311:31) &>' requested here
  191 |         FindSolutionImpl(rank<1>{}, s, context, problem, db_getter, invoke_ctx, perf_cfg, options);
      |         ^
C:/UIF/MIOpen/src/include\miopen/find_solution.hpp:328:25: note: in instantiation of function template specialization 'miopen::solver::FindSolution<TunableTestSolver, miopen::ExecutionContext, TestProblemDescription, (lambda at C:\UIF\MIOpen\src\include\miopen\find_solution.hpp:311:31) &>' requested here
  328 |                         FindSolution(solver, ctx, problem, db, invoke_params, "", std::nullopt);
      |                         ^
C:/UIF/MIOpen/src/include\miopen/find_solution.hpp:292:17: note: in instantiation of function template specialization 'miopen::each_args<(lambda at C:\UIF\MIOpen\src\include\miopen\find_solution.hpp:293:13), TunableTestSolver, RegularTestSolver>' requested here
  292 |         miopen::each_args(
      |                 ^
C:/UIF/MIOpen/src/include\miopen/find_solution.hpp:441:27: note: in instantiation of function template specialization 'miopen::solver::SolverContainer<TunableTestSolver, RegularTestSolver>::SearchForSolutions<TestProblemDescription, miopen::solver::ConvSolution>' requested here
  441 |         const auto slns = SearchForSolutions(ctx, problem, 1, invoke_params);
      |                           ^
C:\UIF\MIOpen\test\gtest\execute_primitive.cpp:201:13: note: in instantiation of function template specialization 'miopen::solver::SolverContainer<TunableTestSolver, RegularTestSolver>::ExecutePrimitive<TestProblemDescription>' requested here
  201 |     solvers.ExecutePrimitive(ctx, problem, miopen::AlgorithmName{"test::algo"}, {});
      |             ^
In file included from C:\UIF\MIOpen\test\gtest\execute_primitive.cpp:28:
In file included from C:/UIF/MIOpen/src/include\miopen/execution_context.hpp:32:
C:/UIF/MIOpen/src/include\miopen/sqlite_db.hpp:474:51: error: no member named 'WhereClause' in 'TestProblemDescription'
  474 |         std::tie(clause, values) = problem_config.WhereClause();
      |                                    ~~~~~~~~~~~~~~ ^
C:/UIF/MIOpen/src/include\miopen/sqlite_db.hpp:646:29: note: in instantiation of function template specialization 'miopen::SQLitePerfDb::FindRecordUnsafe<TestProblemDescription>' requested here
  646 |         const auto record = FindRecordUnsafe(problem_config);
      |                             ^
C:/UIF/MIOpen/src/include\miopen/sqlite_db.hpp:369:50: note: in instantiation of function template specialization 'miopen::SQLitePerfDb::LoadUnsafe<TestProblemDescription, TestPerfConfig>' requested here
  369 |         return reinterpret_cast<Derived*>(this)->LoadUnsafe(args...);
      |                                                  ^
C:/UIF/MIOpen/src/include\miopen/db.hpp:250:18: note: in instantiation of function template specialization 'miopen::SQLiteBase<miopen::SQLitePerfDb>::Load<const TestProblemDescription &, const std::basic_string<char> &, TestPerfConfig &>' requested here
  250 |         if(_user.Load(args...))
      |                  ^
C:/UIF/MIOpen/src/include\miopen/find_solution.hpp:102:26: note: in instantiation of function template specialization 'miopen::DbTimer<miopen::MultiFileDb<miopen::SQLitePerfDb, miopen::SQLitePerfDb, true>>::Load<const TestProblemDescription, const std::basic_string<char>, TestPerfConfig>' requested here
  102 |             else if(db().Load(problem, s.SolverDbId(), config))
      |                          ^
C:/UIF/MIOpen/src/include\miopen/find_solution.hpp:191:9: note: in instantiation of function template specialization 'miopen::solver::FindSolutionImpl<TunableTestSolver, miopen::ExecutionContext, TestProblemDescription, (lambda at C:\UIF\MIOpen\src\include\miopen\find_solution.hpp:311:31) &>' requested here
  191 |         FindSolutionImpl(rank<1>{}, s, context, problem, db_getter, invoke_ctx, perf_cfg, options);
      |         ^
C:/UIF/MIOpen/src/include\miopen/find_solution.hpp:328:25: note: in instantiation of function template specialization 'miopen::solver::FindSolution<TunableTestSolver, miopen::ExecutionContext, TestProblemDescription, (lambda at C:\UIF\MIOpen\src\include\miopen\find_solution.hpp:311:31) &>' requested here
  328 |                         FindSolution(solver, ctx, problem, db, invoke_params, "", std::nullopt);
      |                         ^
C:/UIF/MIOpen/src/include\miopen/find_solution.hpp:292:17: note: in instantiation of function template specialization 'miopen::each_args<(lambda at C:\UIF\MIOpen\src\include\miopen\find_solution.hpp:293:13), TunableTestSolver, RegularTestSolver>' requested here
  292 |         miopen::each_args(
      |                 ^
C:/UIF/MIOpen/src/include\miopen/find_solution.hpp:441:27: note: in instantiation of function template specialization 'miopen::solver::SolverContainer<TunableTestSolver, RegularTestSolver>::SearchForSolutions<TestProblemDescription, miopen::solver::ConvSolution>' requested here
  441 |         const auto slns = SearchForSolutions(ctx, problem, 1, invoke_params);
      |                           ^
C:\UIF\MIOpen\test\gtest\execute_primitive.cpp:201:13: note: in instantiation of function template specialization 'miopen::solver::SolverContainer<TunableTestSolver, RegularTestSolver>::ExecutePrimitive<TestProblemDescription>' requested here
  201 |     solvers.ExecutePrimitive(ctx, problem, miopen::AlgorithmName{"test::algo"}, {});
      |             ^
In file included from C:\UIF\MIOpen\test\gtest\execute_primitive.cpp:28:
In file included from C:/UIF/MIOpen/src/include\miopen/execution_context.hpp:32:
C:/UIF/MIOpen/src/include\miopen/sqlite_db.hpp:479:44: error: no member named 'table_name' in 'TestProblemDescription'
  479 |             "INNER JOIN " + problem_config.table_name() + " "
      |                             ~~~~~~~~~~~~~~ ^
C:/UIF/MIOpen/src/include\miopen/sqlite_db.hpp:554:53: error: no member named 'InsertQuery' in 'TestProblemDescription'
  554 |             std::tie(clause, vals) = problem_config.InsertQuery();
      |                                      ~~~~~~~~~~~~~~ ^
C:/UIF/MIOpen/src/include\miopen/sqlite_db.hpp:361:50: note: in instantiation of function template specialization 'miopen::SQLitePerfDb::UpdateUnsafe<TestProblemDescription, TestPerfConfig>' requested here
  361 |         return reinterpret_cast<Derived*>(this)->UpdateUnsafe(args...);
      |                                                  ^
C:/UIF/MIOpen/src/include\miopen/find_solution.hpp:191:9: note: in instantiation of function template specialization 'miopen::solver::FindSolutionImpl<TunableTestSolver, miopen::ExecutionContext, TestProblemDescription, (lambda at C:\UIF\MIOpen\src\include\miopen\find_solution.hpp:311:31) &>' requested here
  191 |         FindSolutionImpl(rank<1>{}, s, context, problem, db_getter, invoke_ctx, perf_cfg, options);
      |         ^
C:/UIF/MIOpen/src/include\miopen/find_solution.hpp:328:25: note: in instantiation of function template specialization 'miopen::solver::FindSolution<TunableTestSolver, miopen::ExecutionContext, TestProblemDescription, (lambda at C:\UIF\MIOpen\src\include\miopen\find_solution.hpp:311:31) &>' requested here
  328 |                         FindSolution(solver, ctx, problem, db, invoke_params, "", std::nullopt);
      |                         ^
C:/UIF/MIOpen/src/include\miopen/find_solution.hpp:292:17: note: in instantiation of function template specialization 'miopen::each_args<(lambda at C:\UIF\MIOpen\src\include\miopen\find_solution.hpp:293:13), TunableTestSolver, RegularTestSolver>' requested here
  292 |         miopen::each_args(
      |                 ^
C:/UIF/MIOpen/src/include\miopen/find_solution.hpp:441:27: note: in instantiation of function template specialization 'miopen::solver::SolverContainer<TunableTestSolver, RegularTestSolver>::SearchForSolutions<TestProblemDescription, miopen::solver::ConvSolution>' requested here
  441 |         const auto slns = SearchForSolutions(ctx, problem, 1, invoke_params);
      |                           ^
C:\UIF\MIOpen\test\gtest\execute_primitive.cpp:201:13: note: in instantiation of function template specialization 'miopen::solver::SolverContainer<TunableTestSolver, RegularTestSolver>::ExecutePrimitive<TestProblemDescription>' requested here
  201 |     solvers.ExecutePrimitive(ctx, problem, miopen::AlgorithmName{"test::algo"}, {});
      |             ^
In file included from C:\UIF\MIOpen\test\gtest\execute_primitive.cpp:28:
In file included from C:/UIF/MIOpen/src/include\miopen/execution_context.hpp:32:
C:/UIF/MIOpen/src/include\miopen/sqlite_db.hpp:572:53: error: no member named 'WhereClause' in 'TestProblemDescription'
  572 |             std::tie(clause, vals) = problem_config.WhereClause();
      |                                      ~~~~~~~~~~~~~~ ^
C:/UIF/MIOpen/src/include\miopen/sqlite_db.hpp:579:53: error: no member named 'table_name' in 'TestProblemDescription'
  579 |                 "(SELECT id FROM " + problem_config.table_name() +  " "

```

--- a/test/gtest/execute_primitive.cpp
+++ b/test/gtest/execute_primitive.cpp
@@ -34,6 +34,10 @@
 #include <miopen/solver.hpp>
 #include <miopen/temp_file.hpp>
 
+#if MIOPEN_ENABLE_SQLITE && MIOPEN_USE_SQLITE_PERFDB
+#include <miopen/sqlite_db.hpp>
+#endif
+
 #include "get_handle.hpp"
 
 #include <gtest/gtest_common.hpp>
@@ -59,7 +63,12 @@ struct TestProblemDescriptionTag
     std::string updb_path;
 };
 
-struct TestProblemDescription : miopen::ProblemDescriptionBase, TestProblemDescriptionTag
+struct TestProblemDescription : miopen::ProblemDescriptionBase,
+                                TestProblemDescriptionTag
+#if MIOPEN_ENABLE_SQLITE && MIOPEN_USE_SQLITE_PERFDB
+    ,
+                                miopen::SQLiteSerializable<TestProblemDescription>
+#endif
 {
     std::uint32_t net_config;
     TestResults* test_results;
@@ -71,10 +80,27 @@ struct TestProblemDescription : miopen::ProblemDescriptionBase, TestProblemDescr
 
     void Serialize(std::ostream& stream) const { stream << net_config; }
 
+#if MIOPEN_ENABLE_SQLITE && MIOPEN_USE_SQLITE_PERFDB
+    static std::string table_name() { return "config"; }
+#endif
+
+    template <class TSelf>
+    static void Visit(TSelf&& self, std::function<void(std::string, std::string)> visitor)
+    {
+    }
+
+    template <class TSelf>
+    static void Visit(TSelf&& self, std::function<void(std::int64_t, std::string)> visitor)
+    {
+        visitor(static_cast<std::int32_t>(self.net_config), "net_config");
+    }
+
     template <class TSelf, class Visitor>
     static void VisitAll(TSelf&& self, Visitor visitor)
     {
-        visitor(self.net_config, "net_config");
+        Visit(std::forward<Self>(self), [&](int64_t value, std::string name) { f(value, name); });
+        Visit(std::forward<Self>(self),
+              [&](std::string value, std::string name) { f(value, name); });
     }
 
     friend auto GetDb(const miopen::ExecutionContext&, const TestProblemDescriptionTag& problem)

--- a/test/gtest/execute_primitive.cpp
+++ b/test/gtest/execute_primitive.cpp
@@ -98,9 +98,10 @@ struct TestProblemDescription : miopen::ProblemDescriptionBase,
     template <class TSelf, class Visitor>
     static void VisitAll(TSelf&& self, Visitor visitor)
     {
-        Visit(std::forward<Self>(self), [&](int64_t value, std::string name) { f(value, name); });
-        Visit(std::forward<Self>(self),
-              [&](std::string value, std::string name) { f(value, name); });
+        Visit(std::forward<TSelf>(self),
+              [&](int64_t value, std::string name) { visitor(value, name); });
+        Visit(std::forward<TSelf>(self),
+              [&](std::string value, std::string name) { visitor(value, name); });
     }
 
     friend auto GetDb(const miopen::ExecutionContext&, const TestProblemDescriptionTag& problem)

